File: typescript-sdk/examples/mcp-shell.ts
import { HttpAgent } from "../packages/client/src/agent";

async function main() {
  const url = process.env.MCP_SERVER_URL;
  if (!url) {
    console.error("Set MCP_SERVER_URL to the running Smithery server endpoint");
    process.exit(1);
  }

  const agent = new HttpAgent({ url });
  agent.messages = [
    { id: "1", role: "user", content: "hello" },
  ];

  for await (const event of agent.runAgent()) {
    console.log(event);
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: python-sdk/examples/mcp_shell.py
import os
import sys
import json
import requests


def main() -> None:
    url = os.environ.get("MCP_SERVER_URL")
    if not url:
        print("Set MCP_SERVER_URL to the running MCP server endpoint", file=sys.stderr)
        sys.exit(1)

    payload = {
        "messages": [
            {"id": "1", "role": "user", "content": "hello"}
        ]
    }

    with requests.post(url, json=payload, stream=True, headers={"Accept": "text/event-stream"}) as resp:
        resp.raise_for_status()
        for line in resp.iter_lines():
            if not line:
                continue
            try:
                data = json.loads(line.decode("utf-8"))
            except json.JSONDecodeError:
                print(line.decode("utf-8"))
                continue
            print(json.dumps(data, indent=2))


if __name__ == "__main__":
    main()

